src/app/app.component.ts

```diff
@@ -41,14 +47,26 @@ export class AppComponent implements OnInit, OnDestroy {
     this.router.canceledNavigationResolution = 'computed';
     this.store.dispatch([new CheckAuth(), new GetFeaturesList()]);
     this.isWindowMobile(window);
+
+    const savedTheme = localStorage.getItem('preferred-theme');
+    if (savedTheme) {
+      this.isDark = savedTheme === 'dark';
+    } else {
+      this.isDark = window.matchMedia('(prefers-color-scheme: dark)').matches;
```

Comment:
Use the injection token to manipulate the window. I know there are a few other places where we have the same problem, but we can fix them later

Take a look at: https://www.npmjs.com/package/ngx-window-token